addons/mail/static/src/core/common/attachment_list.xml

```diff
@@ -34,7 +34,7 @@
                         <i class="fa fa-spin fa-spinner"/>
                     </div>
                     <div class="position-absolute top-0 bottom-0 start-0 end-0 p-2 text-white opacity-0 opacity-100-hover d-flex align-items-end flax-wrap flex-column">
-                        <div t-if="attachment.isDeletable"
+                        <div t-if="attachment.isDeletable and (env.hasTextContent or props.attachments.length &gt; 1 or env.inChatter or env.inComposer)"
```

Comment:
Hello, @alexkuhn,
Thanks a lot for the review. Sorry for the delay,
I have kept the condition `this.env.hasTextContent` inside the `get showDeleteOnAttachmentCard()` for a specific case. 

- Inside chatter. Send a log note with no TextContent and a single attachment. 
  We Display the `Delete` icon on the attachment inside the AttachmentBox.
  We Do not display the `Delete` icon on the attachment inside the `Message` component.
- Along with checking for text in a message, we also use the condition to check for the presence or absence of the `Message` component.

Here is the link to the image for the above scenario---
[https://www.awesomescreenshot.com/image/42628456?key=93b5bad897c4546f25f0e9e9a5bbd390](url)

Reply:
I would store `message` instead of `hasTextContent` in `env`. That way it becomes `this.env.message?.hasTextContent`. This solves the clarity of expression with `env` (i.e. the attachment is shown in a message with some text content)

Reply:
_Note that env is not reactive, unfortunately... But thankfully, `message` enforces identity of the component `Message`, so it's alright to put in subEnv of `Message` as it remains unchanged for the `Message` component at hand._